packages/i18n/src/types.ts

```diff
@@ -10,3 +16,138 @@ export type TranslatableText< T extends string > = string & {
 	 */
 	readonly __translatableText: T;
 };
+
+/**
+ * Type to extends TanninDomainMetadata to support additional properties.
+ */
+export type I18nDomainMetadata = TanninDomainMetadata & {
+	[ key: string ]: unknown;
+};
+
+/**
+ * Locale data is a record of domain names to their metadata or translations.
+ */
+export type LocaleData = Record< string, I18nDomainMetadata | string[] >;
+
+export type SubscribeCallback = () => void;
+export type UnsubscribeCallback = () => void;
+
+/**
+ * Retrieve the domain to use when calling domain-specific filters.
+ */
+export type getFilterDomain = ( domain?: string ) => string;
+
+/**
+ * An i18n instance
+ */
+export interface I18n< TextDomains extends string = string > {
```

Comment:
Nit: `TextDomain` reads better in documentation.

Reply:
@swissspidy Hi, would appreciate a feedback on this https://github.com/WordPress/gutenberg/issues/68906#issuecomment-3102406258